GH-10083: Migrate ZeroMQ module to Jspecify

anthologia opened the pull request:
Related to: https://github.com/spring-projects/spring-integration/issues/10083

**Changes:**
- Replaced `org.springframework.lang.Nullable` with `org.jspecify.annotations.Nullable`

Should I create separate PRs for each module?

anthologia committed:
GH-10083: Migrate ZeroMQ module to Jspecify

Related to: https://github.com/spring-projects/spring-integration/issues/10083

- Replaced `org.springframework.lang.Nullable` with `org.jspecify.annotations.Nullable`

Signed-off-by: Jooyoung Pyoung <pyoungjy@gmail.com>

artembilan commented:
> Fixes: https://github.com/spring-projects/spring-integration/issues/10083

Is not correct in the PR since it does not fully fix the whole issue.
Please, don't use it for this kind of PRs again.
Better like `Related to: `

> Should I create separate PRs for each module?

Yes, please.
That will be easier for us to review.

Thanks

artembilan reviewed (changes requested):
I'm not sure if this was automated, but can we also look into migrating those `package-info.java` files in this module to the `org.jspecify.annotations.NullMarked` instead of those deprecated over there?
Then it would be great to run `gradlew :spring-integration-zeromq:check` and see if your change passes NullAway.

Thanks

artembilan commented:
I know the `org.jspecify.annotations.NullMarked` is a phase 2 in our issue roadmap, but that is OK to do that for such a small module in one go, so we would know where we are for the rest of the project.

Thanks

artembilan commented:
We have just merged one for Camel module: https://github.com/spring-projects/spring-integration/commit/b07d04f9e5dafe54819ff47ec361a75c6c0e244e.

Feel free to follow its experience!

anthologia committed:
Complete ZeroMQ module migration to JSpecify

- Migrate `package-info.java` files to use `@NullMarked` annotation
- Add `@SuppressWarnings("NullAway.Init")` for fields initialized in lifecycle methods

Signed-off-by: Jooyoung Pyoung <pyoungjy@gmail.com>

anthologia commented:
Thanks for the feedback @artembilan ! 😁

I've updated the PR:
- Migrated all `package-info.java` files to `@NullMarked`
- Added `@SuppressWarnings("NullAway.Init")` for lifecycle-initialized fields
- Updated existing commit message from `Fixes:` to `Related to:` via force push

anthologia committed:
Fix NullAway violations in ZeroMqChannel

- Extract this.zeroMqProxy to local variable inside Mono.defer() block

Signed-off-by: Jooyoung Pyoung <pyoungjy@gmail.com>

anthologia commented:
I've applied the feedback. All tests pass locally, but Kafka module tests are failing in CI. Could you re-trigger the build please?

artembilan commented:
As far as the PR build is green, this is going to be merged.
Thank you!

If we miss something, we can address it later when we got less those warnings in the logs to spot leftovers easier.
Yes, that Kafka test fail sporadically.
have no idea yet why we go same record again when no rebalance.

artembilan commented:
Most of them are related to sockets from libraries we rely on, so it is hard to diagnose.
